src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -0,0 +1,191 @@
+
+export const CsvUploaderProvider: FC<Props> = React.memo(({children}) => {
+  const [value, setValue] = useState(0)
+  const {bucketID} = useParams()
+  const [hasFile, setHasFile] = useState(false)
+  const [uploadFinished, setUploadFinished] = useState(false)
+
+  const bucket =
+    useSelector((state: AppState) =>
+      getByID<Bucket>(state, ResourceType.Buckets, bucketID)
+    )?.name ?? ''
+  const org = useSelector(getOrg)
+
+  const history = useHistory()
+
+  const handleDismiss = useCallback(() => {
+    history.push(`/orgs/${org.id}/load-data/buckets`)
+  }, [history, org.id])
+
+  const handleDrop = useCallback(
+    (csv: string) => {
+      setHasFile(true)
+      setTimeout(() => {
+        const {table} = fromFlux(csv)
+        const filtered = [
+          /^_start$/,
+          /^_stop$/,
+          /^_time$/,
+          /^_value/,
+          /^_measurement$/,
+          /^_field$/,
+          /^table$/,
+          /^result$/,
+        ]
+
+        const columns = table.columnKeys.filter(key => {
+          return filtered.reduce((acc, curr) => {
+            return acc && !curr.test(key)
+          }, true)
+        })
+
+        const length = table.length
+
+        let chunk = ''
+
+        let measurement: any = ''
+        let field: any = ''
+        let time: any = ''
+        let tags: any = ''
+        let line: any = ''
+
+        let counter = 0
+        let progress = 0
+
+        const pendingWrites = []
+        let precision = WritePrecision.Ms
+
+        for (let i = 0; i < length; i++) {
+          if (i !== 0 && i % MAX_CHUNK_SIZE === 0) {
+            // second range
+            if (`${time}`.length === 10) {
+              precision = WritePrecision.S
+            }
+            // millisecond range
+            if (`${time}`.length === 13) {
+              precision = WritePrecision.Ms
+            }
+            // microsecond range
+            if (`${time}`.length === 16) {
+              precision = WritePrecision.Us
+            }
+            // nanosecond range
+            if (`${time}`.length === 19) {
+              precision = WritePrecision.Ns
+            }
+            const resp = postWrite({
+              data: chunk,
+              query: {org: org.name, bucket, precision},
+            }).then(() => {
+              const percent = (++progress / counter) * 100
+              setValue(Math.floor(percent))
+            })
+            pendingWrites.push(resp)
+            counter++
+            chunk = ''
+          }
+          measurement = table.getColumn('_measurement')[i]
+          field = table.getColumn('_field')[i]
+          time = table.getColumn('_time')[i]
+          tags = columns
+            .filter(col => !!table.getColumn(col)[i])
+            .map(col => `${col}=${table.getColumn(col)[i]}`)
+            .join(',')
+            .trim()
+            .replace(/(\r\n|\n|\r)/gm, '')
+          line = `${measurement},${tags} ${field}="${field}" ${time}`
```

Comment:
instead of all this, normalize the time value to nano seconds instead, then always send nanoseconds in the write command
``` line = `${measurement},${tags} ${field}="${field}" ${normalize(time)}` ```

and have normalize do something like ` (time) => time + Array(13 - time.length + 1).join('0')`

Reply:
That's a great idea